serial_server.py

```diff
@@ -53,28 +56,62 @@ class NoPortsAvailableError(Error):
     """No serial ports are available."""
     code = 3
 
+SerialDevice = collections.namedtuple('SerialDevice', ['name', 'devicepath'])
 
 class SerialServer(LabradServer):
     """Provides access to a computer's serial (COM) ports."""
     name = '%LABRADNODE% Serial Server'
 
     def initServer(self):
+        if sys.platform.startswith('win32'):
+            self.enumerate_serial_windows()
+        else:
+            self.enumerate_serial_pyserial()
+            
+    def enumerate_serial_windows(self):
+        """This manually attempts to enumerate the first 20 COM ports.
+        pyserial includes a function to enumerate device names, but it
+        possibly doesn't work right on windows for COM ports above 4.
+        http://stackoverflow.com/questions/12090503/listing-available-com-ports-with-python
+        """
         self.SerialPorts = []
         print 'Searching for COM ports:'
         for a in range(1, 20):
             COMexists = True
             try:
-                ser = Serial('\\\\.\\COM%d' % a)
+                ser = Serial(r'\\.\COM%d' % a)
```

Comment:
also, might be best to pull `r'\\.\'` out as a global like `WIN_DEVICEPATH_PREFIX`